### PR TITLE
Chore/update lint config

### PR DESCRIPTION
**Pull request by ThunderSon**
This PR should be merged before #387 in order to ensure that the documents will render as expected.

- Updated the `markdownlint` tool configuration to properly tackle and handle markdown for mkdocs.
- Added titles and h1s for all files
- Updated the heading levels for all documents
- Fixed all linting issues

Sorry for this file batch :heart:

I was based on nicolas's branch and I didn't force the update when I pulled the main branch. I reverted the commits done by myself and nicolas related to mkdocs in the last commit.

### CODE REVIEW

**Comment by ThunderSon** [comment]
Again, apologies for this fiesta.
A thing that is accepted: comments that were `#` and now are `##`. This is the only regex I used to update the heading levels.
Going through the markdown files is somewhat impossible. I used the linter to fix most of the issues, and some manual updates to lists. Just make sure I didn't throw the project off the cliff by mistake :yum: _Otherwise it's good, I reviewed the work_

**Comment by nicoduj** [comment]
@ThunderSon 

> Added titles and h1s for all files

The problem if we want perfect "TOC" in material is that one sheet should not have more thant 1 H1 (so no more than 1 #) . 

👍 [AJAX_Security_Cheat_Sheet](https://nicoduj.github.io/CheatSheetSeries/cheatsheets/AJAX_Security_Cheat_Sheet/) 
👎 [Abuse_Case_Cheat_Sheet](https://nicoduj.github.io/CheatSheetSeries/cheatsheets/Abuse_Case_Cheat_Sheet/)

**Comment by ThunderSon** [comment]
Haha I am aware of that. It's a rule in the linter as well, all the files have 1 h1 now, the cheat sheet name.

File: cheatsheets/Error_Handling_Cheat_Sheet.md [hunk 1]
```diff
@@ -107,7 +109,7 @@ response.setStatus(200);
 {"message":"An error occur, please retry"}
 ```
 
-## Java SpringMVC/SpringBoot web application
+### Java SpringMVC/SpringBoot web application
```
Comment (kingthorin) [inline cheatsheets/Error_Handling_Cheat_Sheet.md:112]: Can this be made more generic so drop the slash in the middle and simplify linking?

File: cheatsheets/C-Based_Toolchain_Hardening_Cheat_Sheet.md [hunk 2]
```diff
@@ -1,12 +1,14 @@
-# Introduction
+# C-Based Toolchain Hardening Cheat Sheet
```
[the remark this answers is not in the document]
Comment (ThunderSon) [reply cheatsheets/C-Based_Toolchain_Hardening_Cheat_Sheet.md:1]: They will be removed in the upcoming change. Just need to apply the design change and then modify the names and archive the old ones.

File: cheatsheets/DotNet_Security_Cheat_Sheet.md [hunk 3]
```diff
@@ -195,37 +197,37 @@ if ((string)ViewState[AntiXsrfTokenKey] != _antiXsrfTokenValue ||
...
--   If HTTPS is not used, [slidingExpiration](https://docs.microsoft.com/en-us/dotnet/api/system.web.security.formsauthentication.slidingexpiration?view=netframework-4.7.2) should be disabled. Consider disabling [slidingExpiration](https://docs.microsoft.com/en-us/dotnet/api/system.web.security.formsauthentication.slidingexpiration?view=netframework-4.7.2) even with HTTPS.
+- If HTTPS is not used, [slidingExpiration](https://docs.microsoft.com/en-us/dotnet/api/system.web.security.formsauthentication.slidingexpiration?view=netframework-4.7.2) should be disabled. Consider disabling [slidingExpiration](https://docs.microsoft.com/en-us/dotnet/api/system.web.security.formsauthentication.slidingexpiration?view=netframework-4.7.2) even with HTTPS.
 - Always implement proper access controls.
-    - Compare user provided username with `User.Identity.Name`.
-    - Check roles against `User.Identity.IsInRole`.
+  - Compare user provided username with `User.Identity.Name`.
+  - Check roles against `User.Identity.IsInRole`.
```
[the remark this answers is not in the document]
Comment (ThunderSon) [reply cheatsheets/DotNet_Security_Cheat_Sheet.md:215]: 2 spaces, as it should be

File: cheatsheets/Deserialization_Cheat_Sheet.md [hunk 4]
```diff
@@ -285,13 +288,13 @@ If the application knows before deserialization which messages will need to be p
...
-    - [JavaSerialKiller](https://github.com/NetSPI/JavaSerialKiller)
-    - [Java Deserialization Scanner](https://github.com/federicodotta/Java-Deserialization-Scanner)
-    - [Burp-ysoserial](https://github.com/summitt/burp-ysoserial)
-    - [SuperSerial](https://github.com/DirectDefense/SuperSerial)
-    - [SuperSerial-Active](https://github.com/DirectDefense/SuperSerial-Active)
+  - [JavaSerialKiller](https://github.com/NetSPI/JavaSerialKiller)
+  - [Java Deserialization Scanner](https://github.com/federicodotta/Java-Deserialization-Scanner)
+  - [Burp-ysoserial](https://github.com/summitt/burp-ysoserial)
+  - [SuperSerial](https://github.com/DirectDefense/SuperSerial)
+  - [SuperSerial-Active](https://github.com/DirectDefense/SuperSerial-Active)
```
[the remark this answers is not in the document]
Comment (ThunderSon) [reply cheatsheets/Deserialization_Cheat_Sheet.md:295]: they still are, no worries.

File: cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md [hunk 5]
```diff
@@ -102,15 +104,15 @@ Escape the following characters with HTML entity encoding to prevent switching i
  < --> &lt;
  > --> &gt;
  " --> &quot;
- ' --> &#x27;     
- / --> &#x2F;     
+ ' --> &#x27;
+ / --> &#x2F;
```
[the remark this answers is not in the document]
Comment (ThunderSon) [reply cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md:108]: Will be done in a different PR

**Comment by ThunderSon** [comment]
I have reviewed this for the 3rd time. This didn't cause any damage to any CS. I am going to merge this and move this forward.

File: cheatsheets/TLS_Cipher_String_Cheat_Sheet.md [hunk 6]
```diff
@@ -100,25 +102,25 @@ Other option: *Delete this two ciphers from your list*.
...
 
-## Apache
+### Apache
 
 **Cipher String 'B':**
 
 ```text
 SSLProtocol +TLSv1.2                  # for Cipher-String 'A', 'B'
-#SSLProtocol +TLSv1.2 +TLSv1.1 +TLSv1 # for Cipher-String 'C', 'D'
+##SSLProtocol +TLSv1.2 +TLSv1.1 +TLSv1 # for Cipher-String 'C', 'D'
```
Comment (polarathene) [inline cheatsheets/TLS_Cipher_String_Cheat_Sheet.md:119]: Were these meant to have more than one `#`? This and any others in code fences seem to have extra `#`?

I see it referenced by the author in the thread discussion but unclear if they meant they had fixed this in some files or if this was intentional or difficult to correct?

**Pull request merged**